src/openrct2/actions/RideCreateAction.hpp

```diff
@@ -62,64 +63,41 @@ struct RideCreateAction : public GameActionBase<GAME_COMMAND_CREATE_RIDE, GA_FLA
-            for (uint8 *rei = availableRideEntries; *rei != RIDE_ENTRY_INDEX_NULL; rei++)
-            {
-                rideEntry = get_ride_entry(*rei);
+            return std::make_unique<RideCreateGameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_INVALID_RIDE_TYPE);
+        }
 
-                // Can happen in select-by-track-type mode
-                if (!ride_entry_is_invented(*rei) && !gCheatsIgnoreResearchStatus)
-                {
-                    continue;
-                }
 
-                if (!(rideEntry->flags & RIDE_ENTRY_FLAG_SEPARATE_RIDE_NAME) || rideTypeShouldLoseSeparateFlag(rideEntry))
-                {
-                    subType = *rei;
-                    break;
-                }
-            }
-            if (subType == RIDE_ENTRY_INDEX_NULL)
-            {
-                subType = availableRideEntries[0];
-                if (subType == RIDE_ENTRY_INDEX_NULL)
-                {
-                    res->Error = GA_ERROR::INVALID_PARAMETERS;
-                    res->ErrorMessage = STR_INVALID_RIDE_TYPE;
-                    return std::move(res);
-                }
-            }
+        const track_colour_preset_list *colourPresets = &RideColourPresets[rideType];
+        if (colourPreset1 >= colourPresets->count)
+        {
+            // FIXME: Add new error string.
+            return std::make_unique<RideCreateGameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_INVALID_RIDE_TYPE);
         }
 
-        if (subType >= 128)
+        rct_ride_entry *rideEntry = get_ride_entry(rideEntryIndex);
+        vehicle_colour_preset_list *presetList = rideEntry->vehicle_preset_list;
+        if (colourPreset2 >= presetList->count)
         {
-            res->Error = GA_ERROR::INVALID_PARAMETERS;
-            res->ErrorMessage = STR_INVALID_RIDE_TYPE;
-            return std::move(res);
+            // FIXME: Add new error string.
+            return std::make_unique<RideCreateGameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_INVALID_RIDE_TYPE);
         }
 
-        sint32 rideEntryIndex = subType;
+        return std::make_unique<RideCreateGameActionResult>();
+    }
+
+    GameActionResult::Ptr Execute() const override
+    {
+        rct_ride_entry * rideEntry;
+        auto res = std::make_unique<RideCreateGameActionResult>();
+
+        sint32 rideEntryIndex = ride_get_entry_index(rideType, rideSubType);
         sint32 rideIndex = ride_get_empty_slot();
-        if (rideIndex == -1) 
-        {
-            res->Error = GA_ERROR::DISALLOWED;
-            res->ErrorMessage = STR_TOO_MANY_RIDES;
-            return std::move(res);
-        }
 
         res->rideIndex = rideIndex;
-        res->rideColor = ride_get_random_colour_preset_index(rideType) | (ride_get_unused_preset_vehicle_colour(rideType, subType) << 8);
+        //res->rideColor = ride_get_random_colour_preset_index(rideType) | (ride_get_unused_preset_vehicle_colour(rideType, rideEntryIndex) << 8);
```

Comment:
If there's no need for this code, just remove it, don't submit commented out code.